### PR TITLE
planner: fix calculating TiFlash stream count

### PR DESCRIPTION
**Pull request by solotzg**
Signed-off-by: Zhigao Tong <tongzhigao@pingcap.com>

### What problem does this PR solve?

Issue Number: ref #40123

### What is changed and how it works?

- In most x86-64 platforms, `Thread(s) per core` is 2
- ARM cpus don't implement Hyper-threading. `Thread(s) per core` is 1

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/optimizer.go [hunk 1]
```diff
@@ -743,16 +744,29 @@ func calculateTiFlashStreamCountUsingMinLogicalCores(ctx context.Context, sctx s
...
+		}
+		if row[4].GetString() == "cpu-physical-cores" {
+			physicalCores, err := strconv.Atoi(row[5].GetString())
+			if err == nil && physicalCores > 0 {
+				minPhysicalCores = mathutil.Min(minPhysicalCores, uint64(physicalCores))
 			}
 		}
 	}
 	// No need to check len(serersInfo) == serverCount here, since missing some servers' info won't affect the correctness
+	if minPhysicalCores > 1 && minPhysicalCores != initialMaxCores {
```
Comment (yibin87) [inline planner/core/optimizer.go:766]: Not sure if physicalCores are total physical cores or total physical cores used by tiflash server

Comment (solotzg) [reply planner/core/optimizer.go]: OK. 
Use `runtime.GOARCH` to check platforms.

**Review by windtalker — APPROVED** [review]
LGTM

**Comment by yibin87** [comment]
LGTM, however, not tidb reviewer:)

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**